scripts/quests/outlands/You_Call_That_a_Knife.lua

```diff
@@ -0,0 +1,211 @@
+                    end
+                end,
+
+                [128] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED and
+                not player:hasKeyItem(xi.ki.NONBERRYS_KNIFE)
+        end,
+
+        [xi.zone.KAZHAM] =
+        {
+            ['Mhebi_Juhbily'] = quest:event(129):replaceDefault(),
+
+            ['Pula_Rhatti'] = quest:event(269, 0, xi.item.SANDFISH):replaceDefault(),
+
+            ['Vah_Keshura'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(130)
+                    else
+                        return quest:event(131):replaceDefault()
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [130] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end
+            },
+        },
+
+        [xi.zone.TEMPLE_OF_UGGALEPIH] =
+        {
+            ['Chef_Nonberry'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { xi.item.TONBERRY_BOARD }) then
+                        return quest:progressEvent(27)
+                    else
+                        npcUtil.confirmTradedItems(trade) -- Take all items traded
+                        return quest:progressEvent(28)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [27] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    npcUtil.giveKeyItem(player, xi.ki.NONBERRYS_KNIFE)
+                end,
+
+                [28] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    if os.time() >= npc:getLocalVar('cooldown') then
+                        spawnCooks(player, npc)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED and
+                player:hasKeyItem(xi.ki.NONBERRYS_KNIFE)
+        end,
+
+        [xi.zone.KAZHAM] =
+        {
+            ['Mhebi_Juhbily'] = quest:progressEvent(133),
+
+            ['Vah_Keshura'] =  quest:event(132):replaceDefault(),
+
+            onEventFinish =
+            {
+                [133] = function(player, csid, option, npc)
+                    quest:complete(player)
+                    player:delKeyItem(xi.ki.NONBERRYS_KNIFE)
+                    quest:setMustZone(player)
```

Comment:
This is a safe practice to get into. If there's an error when trying to complete the quest, the player won't lose their key item. This is more common and useful with quests that have items in their rewards
```suggestion
                    if quest:complete(player) then
                        player:delKeyItem(xi.ki.NONBERRYS_KNIFE)
                        quest:setMustZone(player)
                    end
```